### PR TITLE
Minor auraTracker improvements

### PR DESCRIPTION
**Pull request by vigo2**
[core] permanent auras are no longer added to activeAuraIDs, speeding up auraTracker.advance() in many cases

### CODE REVIEW

File: sim/core/aura.go [hunk 1]
```diff
@@ -329,124 +324,140 @@ func (at *auraTracker) AddAura(sim *Simulation, newAura Aura) {
...
-	at.auras[newAura.ID].activeIndex = int32(len(at.activeAuraIDs))
-	at.activeAuraIDs = append(at.activeAuraIDs, newAura.ID)
+		// permanent auras can be ignored for advance()
+	} else {
+		newAura.startTime = sim.CurrentTime
+		newAura.expires = sim.CurrentTime + newAura.Duration
+
+		newAura.activeIndex = int32(len(at.activeAuraIDs))
+		at.activeAuraIDs = append(at.activeAuraIDs, newAura.ID)
+	}
```
Comment (jimmyt857) [inline sim/core/aura.go:348]: Looks like you can assign startTime and expires outside of this 'if'.

**Comment by vigo2** [comment]
This came out differently than I thought: it doesn't differentiate between PermanentAuras and Auras, but between auras with or without expiration. So there's no special handling for PermanentAuras.

File: sim/core/aura.go (hunk 1 above)
Comment (vigo2) [reply sim/core/aura.go:348]: Could do for startTime, I thought this is more stupid aka more readable. I accidentally mixed up the order, though.

Comment (vigo2) [reply sim/core/aura.go]: Adressed: changed so it works for similarly to RemoveAura(), and just handles the "Duration != NeverExpires" case. "expires" is never used for such auras, so it remains unset now.

**Pull request merged**